fbw-a32nx/src/systems/fmgc/src/guidance/geometry/GeometryFactory.ts

```diff
@@ -144,13 +144,15 @@ export namespace GeometryFactory {
 
                 const prevLeg = geometry.legs.get(i - 1);
 
-                const oldInboundTransition = geometry.transitions.get(i - 1);
-                const newInboundTransition = TransitionPicker.forLegs(prevLeg, newLeg);
+                if (prevLeg && newLeg) {
```

Comment:
Do you what situations would lead to `prevLeg` or `newLeg` being null here? Just so we don't cover up an underlying issue with a nullcheck.

Reply:
geometry.legs seems to be missing indices in the middle of legs. In my case, when I create a new city pair from EDDM to EHAM, a map is created with legs at key 0 and key 2, but 1 is missing. So as soon as either prevLeg or newLeg is referring to key 1, the error is thrown

Reply:
![image](https://github.com/flybywiresim/aircraft/assets/63071941/182b948e-ac27-4e6e-b9ec-25e0a9064e79)
debug output when i = 0